### PR TITLE
CORE-8047, CORE-7675, CORE-6800 Multimodule project, Notary Support, Simulator Updates

### PR DESCRIPTION
**Pull request by ceebee-dev**
CORE-8047 Multimodule - CSDE project split into workflows and contracts/states subprojects.
CORE-7635 Notary Support - CSDE automated creation of Notary v-node, configurable by config/dev-net.json
CORE-6800 Simulator Updates - updates to the CSDE build files to allows use of the latest Simulator

Other changes:
CORE-8732 Wait for the 'APPROVED' membership registration status instead of 'SUBMITTED'
CORE-8939 
   - 500 response to a v-node creation request is no longer treated as a fatal error.
   - Do not error on CPI upload status 400 response with no Details object.

### CODE REVIEW

**Review by mattbradburyr3 — APPROVED** [review]
Tested: startCorda, deployCordapp, run MyFirstFlow, stop corda. works as expected.
I am accepting that this does not test the multimodule functionality. Merging in anyway in order to be able to start on the refactor. there will be significant tidying up to do after this though.

**Comment by ronanbrowne** [comment]
Your PR gate has failed as this PR introduces a Snyk issue, please see the output of the PR gate [linked ](https://ci02.dev.r3.com/blue/organizations/jenkins/Corda5%2FC5-Samples%2Fcordapp-csde-kotlin/detail/PR-24/1/) for full details, that particular check compares the code in PR to the target branch and scans for any new issues being introduced

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-@Library('corda-shared-build-pipeline-steps@5.0') _
+@Library('corda-shared-build-pipeline-steps@chrisbarratt/CORE-8075-create-pipeline-for-running-the-autotester') _
```
Comment (ronanbrowne) [inline .ci/Jenkinsfile]: I presume this was just for testing and you'll want to change prior to merge ?

is there also a associated corda-shared-build-pipeline-steps PR to merge this into 5.0 branch?

File: buildSrc/src/main/groovy/csde.gradle [hunk 2]
```diff
@@ -162,29 +183,77 @@ tasks.register('createKeystore') {
 }
 
 
-tasks.register('buildCPI') {
+tasks.register('buildCPIs') {
     group = pluginGroupName
-    dependsOn('build', 'createGroupPolicy', 'createKeystore')
+    // dependsOn('contracts:build', 'workflows:build', 'createGroupPolicy', 'createKeystore', 'getNotaryServerCPB')
+
+    //def dependsOnTasks = subprojects.collect {it.tasks.withType(org.gradle.api.tasks.GradleBuild)}
```
Comment (ronanbrowne) [inline buildSrc/src/main/groovy/csde.gradle]: is this and the other commented out code valid , should this be going into a release branch of a public repo ? As this is quite confusing for a dev which comes along trying to use this

File: gradle.properties [hunk 3]
```diff
@@ -35,4 +64,9 @@ devEnvWorkspace=workspace
 dbContainerName=CSDEpostgresql
 
 # R3 internal repository
-artifactoryContextUrl=https://staging.download.corda.net/maven/20ede3c6-29c0-11ed-966d-b7c36748b9f6-Beta1.0-HC00
+#artifactoryContextUrl=https://www.youtube.com
+artifactoryContextUrl=https://staging.download.corda.net/maven/20ede3c6-29c0-11ed-966d-b7c36748b9f6-Fox1.0-RC01
```
Comment (ronanbrowne) [inline gradle.properties:42]: Should we be merging these to the release branch, which is the first branch anyone sees when looking at the repo? 

These URLs are intended for internal testing - seems like these would be better suited to se set in a specific tag build for a given RC and leave a reasonable default in release/*

File: build.gradle [hunk 4]
```diff
@@ -18,128 +18,171 @@ plugins {
...
+    repositories {
+        // All dependencies are held in Maven Central
+        mavenCentral()
+        maven {
+            url = "$artifactoryContextUrl/"
+        }
+
+
+        /*
+    // Internal artifactory repos for dev
```
Comment (ronanbrowne) [inline build.gradle:65]: I know why you've commented these out ( to test the RC) , but i'd question if its correct to have these committed to release branch like this.  Having large amounts of commented out code doesn't look good in our default branch of a public repo IMO.

File: buildSrc/gradle.properties [hunk 5]
```diff
@@ -1,7 +1,17 @@
...
-cordaApiVersion=5.0.0.505-Beta1.0-HC00
 
-# R3 internal repository
-artifactoryContextUrl=https://staging.download.corda.net/maven/20ede3c6-29c0-11ed-966d-b7c36748b9f6-Beta1.0-HC00
+cordaApiVersion=5.0.0.522-Fox1.0-RC01
+#cordaApiVersion=5.0.0.519-Beta-Fox-HC03
+#cordaApiVersion=5.0.0.514-beta-1670584271886
+#cordaApiVersion=5.0.0.508-beta+
+#cordaApiVersion=5.0.0.507-beta-1669967318895
+#cordaApiVersion=5.wrong
```
Comment (ronanbrowne) [inline buildSrc/gradle.properties]: I understand this is for testing purposes but should it be committed to release branch , same with other values here ?

File: buildSrc/src/main/groovy/csde.gradle [hunk 6]
```diff
@@ -1,4 +1,5 @@
 import com.r3.csde.CsdeRpcInterface
+// import net.corda.plugins.cpb2
```
Comment (ronanbrowne) [inline buildSrc/src/main/groovy/csde.gradle:2]: needed?

**Review by ronanbrowne — CHANGES_REQUESTED** [review]
a few comments on the build code, i have not looked at *.kt files

File: buildSrc/src/main/groovy/csde.gradle (hunk 2 above)
Comment (mattbradburyr3) [reply buildSrc/src/main/groovy/csde.gradle]: The whole repo is a bit messy at the moment, this PR is to attempt to get us a working base line to start refactoring it and tidying it all up.

even though its a public repo, its not a branch we expect people to be using yet.

Comment (mattbradburyr3) [reply buildSrc/src/main/groovy/csde.gradle]: This weeks target is to get it all tidied up

File: gradle.properties (hunk 3 above)
Comment (mattbradburyr3) [reply gradle.properties:42]: I think I probably agree with you, we should change our strategy here.

File: buildSrc/src/main/groovy/csde.gradle (hunk 6 above)
Comment (mattbradburyr3) [reply buildSrc/src/main/groovy/csde.gradle:2]: lets have a quick chat - you can stop reviewing for now

File: buildSrc/gradle.properties (hunk 5 above)
Comment (mattbradburyr3) [reply buildSrc/gradle.properties]: Agreed it needs tidying up

File: build.gradle (hunk 4 above)
Comment (mattbradburyr3) [reply build.gradle:65]: agreed

File: build.gradle [hunk 7]
```diff
@@ -7,7 +7,7 @@ plugins {
     // These extend existing build environment so that CPB and CPK files can be built.
     // This includes a CorDapp DSL that allows the developer to supply metadata for the CorDapp
     // required by Corda.
-    id 'net.corda.plugins.cordapp-cpb2'
+    //id 'net.corda.plugins.cordapp-cpb2'
```
Comment (yarivus) [inline build.gradle:10]: please do not leave commented out code

File: contracts/src/main/kotlin/com/r3/developers/csdetemplate/contracts/TestConsensualState.kt [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+import net.corda.v5.ledger.consensual.ConsensualState
+import net.corda.v5.ledger.consensual.transaction.ConsensualLedgerTransaction
+import java.security.PublicKey
+
+class TestConsensualState(
+    val testField: String,
+    override val participants: List<PublicKey>
+) : ConsensualState {
+    override fun verify(ledgerTransaction: ConsensualLedgerTransaction) {}
+}
```
Comment (yarivus) [inline contracts/src/main/kotlin/com/r3/developers/csdetemplate/contracts/TestConsensualState.kt:12]: Don't we use some mocking infrastructure to the interfaces/abstract classes that avoid the definition of such classes as TestConsensualState ?

File: contracts/src/main/kotlin/com/r3/developers/utxodemo/contracts/TestContract.kt [hunk 9]
```diff
@@ -0,0 +1,12 @@
+package com.r3.developers.utxodemo.contracts
+
+import net.corda.v5.ledger.utxo.Command
+import net.corda.v5.ledger.utxo.Contract
+import net.corda.v5.ledger.utxo.transaction.UtxoLedgerTransaction
+
+class TestContract : Contract {
+    override fun verify(transaction: UtxoLedgerTransaction) {
```
Comment (yarivus) [inline contracts/src/main/kotlin/com/r3/developers/utxodemo/contracts/TestContract.kt:8]: see my comment for TestConsensualState

File: contracts/src/main/kotlin/net/cordacon/example/doorcode/contracts/DoorCodeConsensualStates.kt [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+import net.corda.v5.base.annotations.CordaSerializable
+import net.corda.v5.ledger.consensual.ConsensualState
+import net.corda.v5.ledger.consensual.transaction.ConsensualLedgerTransaction
+import java.security.PublicKey
+
+@CordaSerializable
+data class DoorCode(val code: String)
+
+@CordaSerializable
+class DoorCodeConsensualState(val code: DoorCode, override val participants: List<PublicKey>) : ConsensualState {
```
Comment (yarivus) [inline contracts/src/main/kotlin/net/cordacon/example/doorcode/contracts/DoorCodeConsensualStates.kt:12]: See my comment to TestConsensualState

File: gradle.properties [hunk 11]
```diff
@@ -2,10 +2,19 @@ kotlin.code.style=official
 
 # Specify the version of the Corda-API to use.
 # This needs to match the version supported by the Corda Cluster the CorDapp will run on.
-cordaApiVersion=5.0.0.505-Beta1.0-HC00
+cordaApiVersion=5.0.0.522-Fox1.0-RC01
+
+# Settings For Development Utilities
+combinedWorkerVersion=5.0.0.0-Fox1.0-RC01
```
Comment (yarivus) [inline gradle.properties:8]: why do we have ^M end of lines for this file ?

File: settings.gradle [hunk 12]
```diff
@@ -2,9 +2,51 @@ pluginManagement {
...
+            url = "$artifactoryContextUrl/corda-os-maven-unstable-local"
+            credentials {
+                username = settings.ext.find('cordaArtifactoryUsername') ?: System.getenv('CORDA_ARTIFACTORY_USERNAME')
+                password = settings.ext.find('cordaArtifactoryPassword') ?: System.getenv('CORDA_ARTIFACTORY_PASSWORD')
+            }
+        }
+        maven {
+            url = "$artifactoryContextUrl/corda-dev"
+            credentials {
+                username = settings.ext.find('cordaArtifactoryUsername') ?: System.getenv('CORDA_ARTIFACTORY_USERNAME')
```
Comment (yarivus) [inline settings.gradle:43]: see @ronanbrowne 's comment earlier

File: workflows/src/main/kotlin/net/cordapp/demo/obligation/messages/DeleteObligationRequestMessage.kt [hunk 13]
```diff
@@ -0,0 +1,5 @@
+package net.cordapp.demo.obligation.messages
+
+import java.util.UUID
+
+data class DeleteObligationRequestMessage(val id: UUID)
```
Comment (yarivus) [inline workflows/src/main/kotlin/net/cordapp/demo/obligation/messages/DeleteObligationRequestMessage.kt:5]: I think you can create a single file ObligationMessages.kt and create all data classes there

**Review by yarivus — COMMENTED** [review]
see my comments

**Pull request closed without merging**